0004-post-tags.md

```diff
@@ -0,0 +1,324 @@
+
+Another use for tags is better handling of how posts are displayed. For example some people would like the content of a post to stay hidden for all users. This could be the case in a community focused on release discussions of Books, Movies, TV Shows or communities with content some users might find offensive but which is not neccesarily NSFW (for example jokes about certain topics).
+
+In order to ensure this consistency while filtering tags can only be created by privileged users, initially that is intended to be Admins and Community moderators.
+
+Essentially the tags serve as an easier way to message what is *inside* the post. Combining multiple tags allows for even finer control. Made a post about news in poland? Add the "News" and "Poland" tags to it and people interested exclusively in polish news can find it easily or people not interested can avoid it just as easily.
+
+# Reference-level explanation
+
+### Protocol:
+According to https://www.w3.org/TR/activitystreams-vocabulary/#dfn-tags a general tag object exists in the ActivityPub protocol allowing for custom implementations so long as a type is specified.
+To keep things simple an initial community tag object should consist of the following:
+```json
+{
+  id: number,
+  url: string,
+  name: string
+}
+```
+
+Entirely unmoderated tags are not an option for lemmy as the moderation workload would be too much. Additionally users being able to type out tags themselves introduces splintering in the tag contents due to typos. A better solution is a curated list of tags users can attach to their posts. The list of tags can be maintained by both admins and moderators allowing for each community to tailor tags to their specific needs.
+Content for tags would be located in the respective root: `https://example.org/c/example/t/tag`
+
+The tag URL could then be utilized as a unique identifier for the tag, however doing so would restrict the ability to move the underlying tag url in the future. Instead an additional `id` field should be used. Using the object ID optional tag federation can also be achieved, allowing for communities across multiple instances to share content via tags (example: News tag shared across instances). This would also solve the issue of splintered communities across instances while not forcing it on the communities in question. For now tags will only be applicable to posts, however the general design allows for them to be attached to any kind of object later on, be that instance, community or user. The limited initial scope allows for easier modifications should any rough edges or missing features be discovered.
+
+To fill all needed use cases 3 tag flavors will be needed:
+
+- NSFW
+- Content Warning
+- Generic Tag
```

Comment:
```suggestion
```

The rest of the RFC is about community-local post tags, but here introduces global tags suddenly. This doesn't really make sense, since this type of tag would be lemmy-global and not community-local.

If we do want this, we should declare those like this:
```json
{
  "type": "GlobalPostTag",
  "id": "https://join-lemmy.org/tags/NSFW",
  "name": "NSFW"
}
```

But we already have NSFW tag (via the fediverse-compatible "sensitive" flag), so this would really only be useful for other global tags, which sounds out of scope for me here. Makes more sense to (initially) let community moderators decide which tags they want.